src/domains/word/word.domain.ts

```diff
@@ -189,29 +189,37 @@ export class WordDomain extends DomainRoot {
     atd: AccessTokenDomain,
     dto: PutWordByIdBodyDTO,
     wordModel: WordModel,
+    preferenceModel: PreferenceModel,
   ): Promise<WordDomain> {
     if (atd.userId !== this.props.userId) {
       throw new UpdateForbiddenError(atd, `Word`)
     }
-    return WordDomain.fromMdb(
-      await wordModel
-        .findByIdAndUpdate(
-          this.id,
-          {
-            language: dto.languageCode,
-            isFavorite: dto.isFavorite,
-            word: dto.term,
-            pronun: dto.pronunciation,
-            meaning: dto.definition,
-            example: dto.example,
-            exampleLink: dto.exampleLink,
-            tag: dto.tags,
-            isArchived: dto.isArchived,
-          },
-          { new: true },
-        )
-        .exec(),
-    )
+    const updateWord = await wordModel
+      .findByIdAndUpdate(
+        this.id,
+        {
+          language: dto.languageCode,
+          isFavorite: dto.isFavorite,
+          word: dto.term,
+          pronun: dto.pronunciation,
+          meaning: dto.definition,
+          example: dto.example,
+          exampleLink: dto.exampleLink,
+          tag: dto.tags,
+          isArchived: dto.isArchived,
+        },
+        { new: true },
+      )
+      .exec()
+
+    if (dto.tags && dto.tags.length > 0) {
+      const preferenceDomain = await PreferenceDomain.fromMdbByAtd(
+        atd,
+        preferenceModel,
+      )
+      await preferenceDomain.updateWithNewTags(atd, dto.tags, preferenceModel)
```

Comment:
new tags must happen only when new tags are requested.
dto.tags potentially contain original tags the users already have.
original tags that the word already had should not be modified

sample)
Let say there is a word `test` with tag `my-tag`
if you add new tag `new-tag`, the recentTags should be `new-tag` only, not `my-tag`.

Reply:
Handled in [c276338](https://github.com/ajktown/api/pull/114/commits/c276338a329a272cc830e312aa3234a52906f31d)